src/main/java/org/junitpioneer/jupiter/ByteRange.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+public class ByteRange extends Range<Byte> {
+	public ByteRange(byte from, byte to, byte step, boolean closed) {
+		super(from, to, step, closed);
+	}
+
+	@Override
+	public Byte nextValue() {
+		return (byte) (getCurrent() + getStep());
```

Comment:
When changing this to `getCurrent().byteValue() + getStep().byteValue()` (to get rid of the cast), the compiler helpfully pointed out that this may overflow. We need to discuss how to handle that. Wrap around? Or throw an exception?

At fist I thought, this can't happen because when `step`ping from `from` to `to` we'd stop early, but with `from=120`, `to=125`, and `step=10` it _could_ happen, right?

Reply:
TBH, I'm not a big fan of either of these suggestions.
To an average user `ByteRangeSource(from = 120, to = 125, step = 10)` looks like a valid range source (albeit not a particularly elegant one...). We could recognize overflows by comparing the relation between the current and next value to the sign of the step (e.g., if we calculate the next value and overflow, it will be smaller than the current value, which shouldn't happen if the step is positive), and just terminate the stream. Thus, this range source will return a single value, `120`.

I'll add a patch to treat it, and then we can discuss against some code.

Reply:
Just checked, the current implementation would result in a silent wrap, i.e. `120 + 10 ~> -126`. Indeed a corner case we need to discuss.

Reply:
Oh, only now saw the implementation in `hasNext` - good job!